addons/mail/static/src/messaging/component/notification_list/notification_list_notification_group_tests.js

```diff
@@ -0,0 +1,575 @@
+                );
+                assert.strictEqual(
+                    ev.data.action.target,
+                    'current',
+                    "action should have 'current' as target"
+                );
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    'res.partner',
+                    "action should have the group model as res_model"
+                );
+                assert.strictEqual(
+                    JSON.stringify(ev.data.action.domain),
+                    JSON.stringify([['message_has_error', '=', true]]),
+                    "action should have 'message_has_error' as domain"
+                );
+            },
+        },
+    });
+    await this.createNotificationListComponent();
+
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup',
+        "should have 1 notification group"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_counter',
+        "should have 1 group counter"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_counter').textContent.trim(),
+        "(2)",
+        "should have 2 notifications in the group"
+    );
+
+    document.querySelector('.o_NotificationGroup').click();
+    assert.verifySteps(
+        ['do_action'],
+        "should have called do_action correctly"
+    );
+});
+
+QUnit.test('different mail.channel are not grouped', async function (assert) {
+    // `mail.channel` is a special case where notifications are not grouped when
+    // they are linked to different channels, even though the model is the same.
+    assert.expect(6);
+
+    this.data.initMessaging.mail_failures = [{
+        date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+        id: 11,
+        message_type: 'email',
+        model: 'mail.channel', // key element of this test: `mail.channel`
+        model_name: "Channel",
+        notifications: [{
+            failure_type: 'SMTP',
+            id: 21,
+            notification_status: 'exception',
+            notification_type: 'email',
+            partner_id: [41, "Someone"],
+        }],
+        res_id: 31, // key element of this test: a different res_id
+    }, {
+        date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+        id: 12,
+        message_type: 'email',
+        model: 'mail.channel', // key element of this test: `mail.channel`
+        model_name: "Channel",
+        notifications: [{
+            failure_type: 'SMTP',
+            id: 22,
+            notification_status: 'bounce',
+            notification_type: 'email',
+            partner_id: [42, "Someone else"],
+        }],
+        res_id: 32, // key element of this test: a different res_id
+    }];
+    await this.start({
+        hasChatWindow: true, // needed to assert thread.open
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fold') {
+                return;
+            }
```

Comment:
I would make a default mockChannelFold RPC in mail/mock_server, so that this is not necessary here

Reply:
Yes! I will not argue against you when you actually want to factorize stuff in tests.